src/app/store/selectors/VehicleTestResultModel.selectors.ts

```diff
@@ -1,11 +1,43 @@
 import { createSelector, createFeatureSelector } from '@ngrx/store';
-import {IVehicleTestResultModelState} from '../../store/state/VehicleTestResultModel.state';
+import { IVehicleTestResultModelState } from '@app/store/state/VehicleTestResultModel.state';
+import { TestRecordTestType } from '@app/models/test-record-test-type';
+import { TestType } from '@app/models/test.type';
+import { TestResultModel } from '@app/models/test-result.model';
 
 export const selectFeature = createFeatureSelector<IVehicleTestResultModelState>(
   'vehicleTestResultModel'
 );
 
-export const selectSelectedVehicleTestResultModel = createSelector(
+export const getVehicleTestResultModel = createSelector(
+  selectFeature,
+  (state: IVehicleTestResultModelState) => state.vehicleTestResultModel
+);
+
+export const selectTestTypeById = (id: string) =>
+  createSelector(selectFeature, (state: IVehicleTestResultModelState) => {
+    let testType: TestType;
+    let testRecord: TestResultModel;
+
+    if (state.selectedTestResultModel) {
+      testRecord = state.selectedTestResultModel;
+      testRecord.testTypes.some((tTypeRes) => {
+        if (tTypeRes.testNumber === id) {
+          testType = tTypeRes;
+        }
+        return tTypeRes.testNumber === id;
+      });
+    }
+    const test: TestRecordTestType = { testType, testRecord };
```

Comment:
Also this seems to be 2 selectors combined? one for test record and another one for testType as I undestand that we are constructing a new testRecordTestType based on filtered testTypes by Id and TestResultModel
Would need @alleypa eyes on that as well?
I think we can combine selectors to retrieves different slices of state and then use maps to reshape it the way we want.
What do you think?